Tools/environment_install/installROS-ubuntu.sh

```diff
@@ -0,0 +1,230 @@
+    PIP=pip3
+    heading "${green}Detected RPi Buster, installing ROS Melodic${reset}"
+elif [ ${RELEASE_CODENAME} == 'focal' ]; then
+    #Ubuntu 20.04 - Noetic
+    ROS_VERSION="noetic"
+    PYTHON_V="python3"
+    PIP=pip3
+    heading "${green}Detected Ubuntu 20.04, installing ROS Noetic${reset}"
+elif [ ${RELEASE_CODENAME} == 'jammy' ]; then
+    #Ubuntu 22.04 - unsupported only ROS2
+    heading "${red}Currently only ROS1 is supported. This Ubuntu release can only be used with ROS2.${reset}"
+    exit 1
+else
+    # We assume an unsupported OS is being used.
+    heading "${red}Unsupported OS detected. Please refer to the ROS webpage to find how to install ROS1 on your system if at all possible.${reset}"
+    exit 1
+fi
+
+# Check to see if other packages were specified
+# If not, set the default base package
+if [ ${#packages[@]}  -eq 0 ] ; then
+ packages+="desktop-full"
+fi
+echo "Packages to install: "${packages[@]}
+# Check to see if we have a ROS base kinda thingie
+hasBasePackage=false
+for package in "${packages[@]}"; do
+  if [[ $package == "ros-base" ]]; then
+     packages=" ros-${ROS_VERSION}-ros-base"
+     hasBasePackage=true
+     break
+  elif [[ $package == "desktop" ]]; then
+     packages=" ros-${ROS_VERSION}-desktop"    
+     hasBasePackage=true
+     break
+  elif [[ $package == "desktop-full" ]]; then
+     packages=" ros-${ROS_VERSION}-desktop-full"    
+     hasBasePackage=true
+     break
+  fi
+done
+if [ $hasBasePackage == false ] ; then
+   shouldInstallPackages
+   exit 1
+fi
+
+heading "${green}Adding Repositories and source lists${reset}"
+#Lets start instaling stuff
+sudo apt-add-repository universe
+sudo apt-add-repository multiverse
+sudo apt-add-repository restricted
+
+# Setup sources.lst
+sudo sh -c 'echo "deb http://packages.ros.org/ros/ubuntu $(lsb_release -sc) main" > /etc/apt/sources.list.d/ros-latest.list'
+# Setup keys
+sudo apt-key adv --keyserver 'hkp://keyserver.ubuntu.com:80' --recv-key C1CF6E31E6BADE8868B172B4F42ED6FBAB17C654
+# If you experience issues connecting to the keyserver, you can try substituting hkp://pgp.mit.edu:80 or hkp://keyserver.ubuntu.com:80 in the previous command.
+# Installation
+
+heading "${green}Updating apt${reset}"
+sudo apt update
+
+heading "${green}Installing ROS${reset}"
+# Here we loop through any packages passed on the command line
+# Install packages ...
+for package in "${packages[@]}"; do
+  sudo apt install $package -y
+done
+
+# This is where you might start to modify the packages being installed, i.e.# sudo apt install ros-${ROS_VERSION}-{package_name}
+# sudo apt install -y setpriv
+# sudo apt install -y ros-${ROS_VERSION}-robot-upstart
+# sudo apt install -y ros-${ROS_VERSION}-navigation
+
+# install other needed packages
+sudo apt install build-essential cmake -y
+
+#
+# To find available packages:
+# apt-cache search ros-melodic
+# 
+# Initialize rosdep
+
+heading "${green}Installing rosdep${reset}"
+
+sudo apt install python3-rosdep -y
```

Comment:
if you  are on python2 distro this need to pull the whole python3 stack

Reply:
That's what PYTHON_V is for.